### PR TITLE
Support for additional claims in the JWT tokens, needed for Google's Identity-Aware Proxy

### PR DESCRIPTION
**Pull request by kalugny**
Hi and thanks for an excellent package!

We're using Google Cloud's Identity-Aware Proxy (IAP) for which `httr` is _almost_ compatible.
The only discrepancy is that IAP requires an additional "claim" on the JWT.

This pull requests:
1.  Adds support for additional claims on the JWT, which you can pass as additional parameters to `oauth_service_token`
2. Supports getting the authentication header from the `id_token` if `access_token` is not set (which is the case for IAP tokens)

Using this for IAP is now very simple:
```r
token <- oauth_service_token(
  endpoint = oauth_endpoints("google"),
  secrets = jsonlite::fromJSON('/path/to/credentials.json'),
  target_audience = iap_oauth_client_id
)
```

### CODE REVIEW

**Comment by kalugny** [comment]
This is the google docs for it:
https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_service_account

**Comment by kalugny** [comment]
@hadley any comments? Should I change something?
It's a simple extension that would support more use-cases.

File: R/oauth-server-side.R [hunk 1]
```diff
@@ -42,6 +43,7 @@ init_oauth_service_account <- function(secrets, scope = NULL, sub = NULL) {
 #'   00:00:00 UTC, January 1, 1970. This value has a maximum of 1 hour from
 #'   the issued time.
 #' @param duration Duration of token, in seconds.
+#' @param ... any additional claims for the token.
```
Comment (hadley) [inline R/oauth-server-side.R:46]:
```suggestion
#' @param ... Any additional claims for the token.
```

File: R/oauth-token.r [hunk 2]
```diff
@@ -301,6 +301,7 @@ Token2.0 <- R6::R6Class("Token2.0", inherit = Token, list(
 #' @inheritParams oauth2.0_token
 #' @inheritParams jwt_signature
 #' @param secrets Secrets loaded from JSON file, downloaded from console.
+#' @param ... any additional claims for the token
```
Comment (hadley) [inline R/oauth-token.r:304]: I think you can remove this line since we're already inheriting docs from `jwt_signature`?

File: R/oauth-token.r [hunk 3]
```diff
@@ -311,7 +312,7 @@ Token2.0 <- R6::R6Class("Token2.0", inherit = Token, list(
 #'
 #' token <- oauth_service_token(endpoint, secrets, scope)
 #' }
-oauth_service_token <- function(endpoint, secrets, scope = NULL, sub = NULL) {
+oauth_service_token <- function(endpoint, secrets, scope = NULL, sub = NULL, ...) {
```
Comment (hadley) [inline R/oauth-token.r:315]: Using `...` here seems a bit broad. I'd think it should be `claims` or similar?

**Comment by hadley** [comment]
httr has been superseded in favour of httr2, so is no longer under active development. Thanks for using httr, thanks for contributing, and my apologies that your contribution never made it into the package.

**Pull request closed without merging**